lib/srv/sshserver.go

```diff
@@ -878,7 +878,7 @@ func (s *Server) handleAgentForward(ch ssh.Channel, req *ssh.Request, ctx *ctx)
 	ctx.setAgent(clientAgent, authChan)
 
 	pid := os.Getpid()
-	socketPath := filepath.Join(os.TempDir(), fmt.Sprintf("teleport-agent-%v.socket", uuid.New()))
+	socketPath := filepath.Join(os.TempDir(), fmt.Sprintf("teleport-agent-%v.socket", uuid.New()[0:8]))
```

Comment:
Two other options:

1. We create the socket under `/var/run/teleport/`. This way our socket files are on the same place on all platforms. We can name the socket itself `teleport-{random number}-{pid}.socket` and still be under the 104 character limit.

2. We do this instead: `ioutil.TempDir(os.TempDir(), "teleport-")` which will create `/tmp/teleport-699583643` on Linux and `/var/folders/00/l7kftvrn2tx7cv0_w2x_c4_c0000gn/T/teleport-699583643 ` on macOS. We can name the socket itself `teleport-{pid}.socket`.